pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx

```diff
@@ -0,0 +1,205 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import React, { useContext, useEffect, useState } from 'react'
+import {
+  UiInstruction,
+  DualFinanceStakingOptionForm,
+} from '@utils/uiTypes/proposalCreationTypes'
+import { NewProposalContext } from '../../../new'
+import GovernedAccountSelect from '../../GovernedAccountSelect'
+import { Governance } from '@solana/spl-governance'
+import useGovernanceAssets from '@hooks/useGovernanceAssets'
+import { ProgramAccount } from '@solana/spl-governance'
+import Input from '@components/inputs/Input'
+import getConfigInstruction from '@utils/instructions/Dual'
+import useWalletStore from 'stores/useWalletStore'
+import { getDualFinanceStakingOptionSchema } from '@utils/validations'
+import Tooltip from '@components/Tooltip'
+
+const StakingOption = ({
+  index,
+  governance,
+}: {
+  index: number
+  governance: ProgramAccount<Governance> | null
+}) => {
+  const [form, setForm] = useState<DualFinanceStakingOptionForm>({
+    soName: undefined,
+    optionExpirationUnixSeconds: 0,
+    numTokens: 0,
+    lotSize: 0,
+    baseTreasury: undefined,
+    quoteTreasury: undefined,
+    payer: undefined,
+    userPk: undefined,
+    strike: 0,
+  })
+  const connection = useWalletStore((s) => s.connection)
+  const wallet = useWalletStore((s) => s.current)
+  const shouldBeGoverned = !!(index !== 0 && governance)
+  const { governedTokenAccountsWithoutNfts } = useGovernanceAssets()
+  const [governedAccount, setGovernedAccount] = useState<
+    ProgramAccount<Governance> | undefined
+  >(undefined)
+
+  const [formErrors, setFormErrors] = useState({})
+  const { handleSetInstructions } = useContext(NewProposalContext)
+  const handleSetForm = ({ propertyName, value }) => {
+    setFormErrors({})
+    setForm({ ...form, [propertyName]: value })
+  }
+  async function getInstruction(): Promise<UiInstruction> {
+    const obj: UiInstruction = await getConfigInstruction({
+      connection,
+      form,
+      schema,
+      setFormErrors,
+      wallet,
+    })
+    return obj
+  }
```

Comment:
nit: remove async wrapping

```suggestion
  function getInstruction(): Promise<UiInstruction> {
    return getConfigInstruction({
      connection,
      form,
      schema,
      setFormErrors,
      wallet,
    })
  }
```